File: sandbox-runner/patterns/provider-fallback/python/starter.py
import asyncio
import time

from temporalio.client import Client

from shared import TASK_QUEUE, WORKFLOW_ID_PREFIX
from workflows import ProviderFallbackWorkflow


async def main() -> None:
    client = await Client.connect("localhost:7233")

    workflow_id = f"{WORKFLOW_ID_PREFIX}-{int(time.time() * 1000)}"
    # Change to "" (empty) to exercise the abort (invalid request) path.
    question = "What is the meaning of durable execution?"

    handle = await client.start_workflow(
        ProviderFallbackWorkflow.run,
        question,
        id=workflow_id,
        task_queue=TASK_QUEUE,
    )
    print(f"Started workflow: {workflow_id}")

    result = await handle.result()
    print(result)
    print(
        f"Open the Temporal UI and search for '{workflow_id}' to see the agent loop and provider sweep."
    )


if __name__ == "__main__":
    asyncio.run(main())

File: sandbox-runner/patterns/provider-fallback/typescript/starter.ts
import { Client, Connection } from "@temporalio/client";

import { TASK_QUEUE, WORKFLOW_ID_PREFIX } from "./shared";
import { providerFallbackWorkflow } from "./workflows";

async function main(): Promise<void> {
  const connection = await Connection.connect();
  try {
    const client = new Client({ connection });
    const workflowId = `${WORKFLOW_ID_PREFIX}-${Date.now()}`;
    // Change to "" (empty) to exercise the abort (invalid request) path.
    const question = "What is the meaning of durable execution?";

    const handle = await client.workflow.start(providerFallbackWorkflow, {
      args: [question],
      taskQueue: TASK_QUEUE,
      workflowId,
    });
    console.log(`Started workflow: ${workflowId}`);

    const result = await handle.result();
    console.log(result);
    console.log(`Open the Temporal UI and search for '${workflowId}' to see the agent loop and provider sweep.`);
  } finally {
    await connection.close();
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
